packages/node/__tests__/integration.spec.ts

```diff
@@ -851,6 +853,157 @@ it('should expose Node req and res objects in the context', async () => {
+        return Promise.resolve({ done: false, value: 'a' })
+      }
+      return p
+    },
+    return: jest.fn(() => Promise.resolve({ done: true, value: undefined })),
+  }
+  const server = createServer({
+    logging: false,
+    schema: {
+      typeDefs: /* GraphQL */ `
+        type Query {
+          _: Boolean
+        }
+        type Subscription {
+          foo: String
+        }
+      `,
+      resolvers: {
+        Subscription: {
+          foo: {
+            resolve: () => 'bar',
+            subscribe: () => fakeIterator,
+          },
+        },
+      },
+    },
+  })
+  try {
+    await server.start()
+
+    // Start and Close a HTTP SSE subscription
+    await new Promise<void>((res) => {
+      const eventSource = new EventSource(
+        `${server.getServerUrl()}?query=subscription{foo}`,
+      )
+      eventSource.onmessage = (ev) => {
+        eventSource.close()
+        res()
+      }
+    })
+    resolve()
+
+    // very small timeout to make sure the subscription is closed
+    await new Promise((res) => setTimeout(res, 30))
+    expect(fakeIterator.return).toHaveBeenCalled()
+  } finally {
+    await server.stop()
+  }
+})
+
+test.only('defer/stream is closed properly', async () => {
+  let counter = 0
+  let resolve: () => void = () => {
+    throw new Error('Noop')
+  }
+
+  const p = new Promise<IteratorResult<ExecutionResult>>((res) => {
+    resolve = () => res({ done: true, value: { data: 'end' } })
+  })
+
+  const fakeIterator: AsyncIterableIterator<ExecutionResult> = {
+    [Symbol.asyncIterator]: () => fakeIterator,
+    next: () => {
+      console.log('LETS GOOO')
+      if (counter === 0) {
+        counter = counter + 1
+        return Promise.resolve({
+          done: false,
+          value: { data: 'turtles' },
+        } as any)
+      }
+      return p
+    },
+    return: jest.fn(() => Promise.resolve({ done: true, value: undefined })),
+  }
+  const plugin: Plugin = {
+    onExecute(ctx) {
+      ctx.setExecuteFn(() => Promise.resolve(fakeIterator) as any)
+    },
+    /* skip validation :) */
+    onValidate(ctx) {
+      ctx.setValidationFn(() => [])
+    },
+  }
+
+  const server = createServer({
+    logging: false,
+    plugins: [plugin],
+  })
+
+  try {
+    await server.start()
+    console.log(server.getServerUrl())
+    const abort = new AbortController()
+    const res = await fetch(server.getServerUrl(), {
+      method: 'POST',
+      headers: {
+        'content-type': 'application/json',
+        accept: 'multipart/mixed',
+      },
+      body: JSON.stringify({
+        query: /* GraphQL */ `
+          query {
+            a
+          }
+        `,
+      }),
+      signal: abort.signal,
+    })
+
+    // Start and Close a HTTP Request
+    const iterator = res.body![Symbol.asyncIterator]()
+    let iteratorResult: IteratorResult<Uint8Array>
+    while ((iteratorResult = await iterator.next())) {
+      console.log(iteratorResult.value.toString())
```

Comment:
Unrelated to the issue in this PR: This causes different console.log results on Node.js 14 and Node.js 16

node.js 14:

```
    console.log
      ---
      Content-Type: application/json; charset=utf-8
      Content-Length: 18
      
      {"data":"turtles"}
      ---

      at Object.log (packages/node/__tests__/integration.spec.ts:985:15)
```

Node.js 16:

```
    console.log
      45,45,45,13,10,67,111,110,116,101,110,116,45,84,121,112,101,58,32,97,112,112,108,105,99,97,116,105,111,110,47,106,115,111,110,59,32,99,104,97,114,115,101,116,61,117,116,102,45,56,13,10,67,111,110,116,101,110,116,45,76,101,110,103,116,104,58,32,49,56,13,10,13,10,123,34,100,97,116,97,34,58,34,116,117,114,116,108,101,115,34,125,13,10,45,45,45
      at Object.log (packages/node/__tests__/integration.spec.ts:985:15)
```

Reply:
on Node 14 this is Buffer
on Node 16 this is Uint8Array

Reply:
Worked around this bug in https://github.com/dotansimha/graphql-yoga/pull/1399/commits/6671a426db67659e6a61b27b360423fd1dc04655

Reply:
I used the reader interface in https://github.com/dotansimha/graphql-yoga/pull/1399/commits/9d55a1e96f516451ac928d1ffe363682e350ff28, however, this is still something that should be addressed (?)